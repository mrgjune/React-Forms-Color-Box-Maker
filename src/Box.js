import React from 'react';
class Box extends React.Component {
    constructor(props) {
        super(props)

        this.handleRemove = this.handleRemove.bind(this);
    }
    /**when button is clicked pass the box ID to the remove prop */
    handleRemove(evt) {
        console.log(this.props)
        this.props.remove(this.props.id);
    }

    /**return a box*/
    render() {
        const { width, height, color } = this.props;

        return (
            <div style={{
                height: `${height}em`, width: `${width}em`, backgroundColor: color
            }} >
                <button onClick={this.handleRemove}>X</button>

            </div>

        )
    }
}

export default Box;