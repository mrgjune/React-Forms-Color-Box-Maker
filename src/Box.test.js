import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Box from "./Box";

describe('check that box renders correctly', () => {
  let wrapper = shallow (<Box/> );
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
