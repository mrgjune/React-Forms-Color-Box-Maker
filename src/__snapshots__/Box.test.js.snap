// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check that box renders correctly renders correctly 1`] = `ShallowWrapper {}`;
